### ISSUE TITLE
Pre-trained model of A2MIM

### ISSUE THREAD

**Issue by zugofn** [Westlake-AI/openmixup#13]
Hi, when will the pre-trained model of A2MIM be released in the [webpage](https://github.com/Westlake-AI/openmixup/tree/main/configs/selfsup/a2mim/)?
The work is so meaningful！
Thank you!

**Comment by Lupin1998** [comment]
Hi, @zugofn. Thanks for your appreciation of A2MIM. Since A2MIM is under review, we plan to release the pre-trained model in late September or later. We will inform you when we update the release. Thanks again for your attention and patience!

**Comment by zugofn** [comment]
> Hi, @zugofn. Thanks for your appreciation of A2MIM. Since A2MIM is under review, we plan to release the pre-trained model in late September or later. We will inform you when we update the release. Thanks again for your attention and patience!

Thanks for your quick reply！
Have a nice day😊

**Comment by donggoing** [comment]
> Hi, @zugofn. Thanks for your appreciation of A2MIM. Since A2MIM is under review, we plan to release the pre-trained model in late September or later. We will inform you when we update the release. Thanks again for your attention and patience!

Is it ok to release the model parameters now?  I would like to do research based on this paper.

**Comment by Lupin1998** [comment]
> > Hi, @zugofn. Thanks for your appreciation of A2MIM. Since A2MIM is under review, we plan to release the pre-trained model in late September or later. We will inform you when we update the release. Thanks again for your attention and patience!
> 
> Is it ok to release the model parameters now? I would like to do research based on this paper.

Hi, @donggoing. Thanks for your attention. Unfortunately, the pre-trained models cannot be released now because we are working on the revision and resubmission of A2MIM. We are sorry about the late release and we plan to upload the pre-trained models after the deadline for ICLR2023 submission.

Meanwhile, feel free to contact me if you have some questions about A2MIM. We can also discuss the relevant questions encountered in your research.

**Comment by donggoing** [comment]
> > > Hi, @zugofn. Thanks for your appreciation of A2MIM. Since A2MIM is under review, we plan to release the pre-trained model in late September or later. We will inform you when we update the release. Thanks again for your attention and patience!
> > 
> > 
> > Is it ok to release the model parameters now? I would like to do research based on this paper.
> 
> Hi, @donggoing. Thanks for your attention. Unfortunately, the pre-trained models cannot be released now because we are working on the revision and resubmission of A2MIM. We are sorry about the late release and we plan to upload the pre-trained models after the deadline for ICLR2023 submission.
> 
> Meanwhile, feel free to contact me if you have some questions about A2MIM. We can also discuss the relevant questions encountered in your research.

Thanks for your quick reply! ”after the deadline for ICLR2023 submission“ you mean like after Sep 28? 
And i would like to know how do you think about what brings the gap between Tr（A2MIM） and CNN（A2MIM）?

**Comment by Lupin1998** [comment]
Yes, @donggoing, it might be uploaded in early October. As for your question, A2MIM works for Transformer and CNN, and there is no difference in its algorithm for both two architectures. When Transformer and CNN adopt A2MIM using the same pre-training epochs, the performance gap might come from two aspects:

(1) Transformer architectures are more suitable for MIM pre-training than CNNs. I have pre-trained A2MIM for longer epochs based on both architectures, ViT-B obtains slight performance gains (800 epochs vs. 1200 epochs), while ResNet-50 suffers from performance degeneration (300 epochs vs. 600 epochs). From my perspective, CNNs have more substantial inductive bias than Transformers but have fewer degrees of freedom. MIM empowerer Transformers with the relevant inductive bias, which helps Transformers to achieve better fine-tuning performances.

(2) The design gap of architecture between ResNet-50 and ViT might cause the sub-optimal performances of CNNs. As you know, ViT belongs to [MetaFormer](https://arxiv.org/abs/2111.11418) architectures, which is more powerful than ResNet (the vanilla bottleneck module). In A2MIM, we only verify ResNet-50 and ViT architectures. If you adopt advanced ConvNet architectures (e.g., [ConvNeXt](https://arxiv.org/abs/2201.03545)), you might find Transformers and CNNs yield similar pre-training performances.

**Comment by donggoing** [comment]
@Lupin1998 I got another question, how do a2mim mask exactly? does it mask the original img? I don't know how the openmixup's pipeline works, so when does 'BlockwiseMaskGenerator' is applied, which confued me.

**Comment by ksnzh** [comment]
Waiting for pre-trained weights.

**Comment by Lupin1998** [comment]
Hi guys, thanks for your attention and patience. we have uploaded the release of weights, logs, visualizations of [A2MIM](https://arxiv.org/abs/2205.13943) and relevant self-supervised methods in [a2mim-in1k-weights](https://github.com/Westlake-AI/openmixup/releases/tag/a2mim-in1k-weights). We are sorry for the delay and hope you will find it useful. Feel free to ask me if you have further questions.

**Comment by ksnzh** [comment]
@Lupin1998 Thanks for your work. I tried to find the self-supervised pre-training weight of a2mim without finetune. Does [full_a2mim_r50_l3_sz224_init_8xb256_cos_ep100](https://github.com/Westlake-AI/openmixup/releases/download/a2mim-in1k-weights/full_a2mim_r50_l3_sz224_init_8xb256_cos_ep100.pth) stand for that?

**Comment by Lupin1998** [comment]
> @Lupin1998 Thanks for your work. I tried to find the self-supervised pre-training weight of a2mim without finetune. Does [full_a2mim_r50_l3_sz224_init_8xb256_cos_ep100](https://github.com/Westlake-AI/openmixup/releases/download/a2mim-in1k-weights/full_a2mim_r50_l3_sz224_init_8xb256_cos_ep100.pth) stand for that?

Yes, weights starting with `full` contain the full weights of pre-training models, while those with `backbone` are the backbone-only weights (can be used for downstream tasks in PyTorch implementations).

**Comment by Lupin1998** [comment]
Hi, guys @zugofn, @donggoing, @ksnzh. Since we have uploaded the release [a2mim-in1k-weights](https://github.com/Westlake-AI/openmixup/releases/tag/a2mim-in1k-weights) and updated documents in issue #31, we will close this issue in two days if there are no new questions. You can reopen it or start a new issue if you have more questions. I am pleased to discuss research topics relevant to self-supervised learning with you! 😃

**Comment by donggoing** [comment]
@Lupin1998 Ok~Thanks for your work！

**Comment by zugofn** [comment]
> Hi, guys @zugofn, @donggoing, @ksnzh. Since we have uploaded the release [a2mim-in1k-weights](https://github.com/Westlake-AI/openmixup/releases/tag/a2mim-in1k-weights) and updated documents in issue #31, we will close this issue in two days if there are no new questions. You can reopen it or start a new issue if you have more questions. I am pleased to discuss research topics relevant to self-supervised learning with you! 😃

Thanks for your quick and patient reply, thank you for your work!
Happy weekend~😀

**Comment by Lupin1998** [comment]
I closed this issue if there is no more question. You can reopen it or start a new issue if you have more questions. Thanks again for your attention.
